File: src/main/java/dev/jsinco/luma/lumaitems/enums/RomanNumeral.java
package dev.jsinco.luma.lumaitems.enums;

// https://stackoverflow.com/questions/3921866/how-do-you-find-a-roman-numeral-equivalent-of-an-integer
public enum RomanNumeral {
    I(1),
    IV(4),
    V(5),
    IX(9),
    X(10),
    XL(40),
    L(50),
    XC(90),
    C(100),
    CD(400),
    D(500),
    CM(900),
    M(1000);

    final int weight;

    RomanNumeral(int weight) {
        this.weight = weight;
    }

    public static String fromInt(int n) {
        if (n <= 0) {
            throw new IllegalArgumentException();
        }

        StringBuilder buf = new StringBuilder();

        final RomanNumeral[] values = RomanNumeral.values();
        for (int i = values.length - 1; i >= 0; i--) {
            while (n >= values[i].weight) {
                buf.append(values[i]);
                n -= values[i].weight;
            }
        }
        return buf.toString();
    }

}

File: src/main/java/dev/lumas/lumaitems/enums/RomanNumeral.kt
package dev.lumas.lumaitems.enums

// https://stackoverflow.com/questions/3921866/how-do-you-find-a-roman-numeral-equivalent-of-an-integer
enum class RomanNumeral(val weight: Int) {
    I(1),
    IV(4),
    V(5),
    IX(9),
    X(10),
    XL(40),
    L(50),
    XC(90),
    C(100),
    CD(400),
    D(500),
    CM(900),
    M(1000);

    companion object {
        fun fromInt(n: Int): String {
            var n = n
            require(n > 0)

            val buf = StringBuilder()

            val values = RomanNumeral.entries.toTypedArray()
            for (i in values.indices.reversed()) {
                while (n >= values[i].weight) {
                    buf.append(values[i])
                    n -= values[i].weight
                }
            }
            return buf.toString()
        }
    }
}
